l2-contracts/contracts/ConsensusRegistry.sol

```diff
@@ -0,0 +1,278 @@
+    /// @param _weight The new attester weight to assign to the node.
+    function changeAttesterWeight(address _nodeOwner, uint256 _weight) external onlyOwner {
+        verifyNodeOwnerExists(_nodeOwner);
+        nodes[_nodeOwner].attesterWeight = _weight;
+    }
+
+    /// @notice Changes the validator's public key and proof-of-possession (PoP) in the registry.
+    /// @dev Only callable by the contract owner or the node owner.
+    /// @dev Verifies that the node owner exists in the registry.
+    /// @param _nodeOwner The address of the node's owner whose validator key and PoP will be changed.
+    /// @param _pubKey The new BLS12-381 public key to assign to the node's validator.
+    /// @param _pop The new proof-of-possession (PoP) to assign to the node's validator.
+    function changeValidatorKey(
+        address _nodeOwner,
+        bytes calldata _pubKey,
+        bytes calldata _pop
+    ) external onlyOwnerOrNodeOwner(_nodeOwner) {
+        require(_pubKey.length > 0, "_pubKey cannot be empty");
+        require(_pop.length > 0, "_pop cannot be empty");
+        verifyNodeOwnerExists(_nodeOwner);
+        nodes[_nodeOwner].validatorPubKey = _pubKey;
+        nodes[_nodeOwner].validatorPoP = _pop;
+    }
+
+    /// @notice Changes the attester's public key of a node in the registry.
+    /// @dev Only callable by the contract owner or the node owner.
+    /// @dev Verifies that the node owner exists in the registry.
+    /// @param _nodeOwner The address of the node's owner whose attester public key will be changed.
+    /// @param _pubKey The new ECDSA public key to assign to the node's attester.
+    function changeAttesterPubKey(
+        address _nodeOwner,
+        bytes calldata _pubKey
+    ) external onlyOwnerOrNodeOwner(_nodeOwner) {
+        require(_pubKey.length > 0, "_pubKey cannot be empty");
+        verifyNodeOwnerExists(_nodeOwner);
+        nodes[_nodeOwner].attesterPubKey = _pubKey;
+    }
+
+    /// @notice Rotates the validator committee list based on active nodes in the registry.
+    /// @dev Only callable by the contract owner.
+    function setValidatorCommittee() external onlyOwner {
+        // Creates a new committee based on active validators.
+        delete validatorCommittee;
+        uint256 len = nodeOwners.length;
+        for (uint256 i = 0; i < len; ++i) {
+            address nodeOwner = nodeOwners[i];
+            Node memory node = nodes[nodeOwner];
+            if (node.active) {
+                validatorCommittee.push(
+                    CommitteeValidator(nodeOwner, node.validatorWeight, node.validatorPubKey, node.validatorPoP)
+                );
+            }
+        }
+    }
+
+    /// @notice Rotates the attester committee list based on active nodes in the registry.
+    /// @dev Only callable by the contract owner.
+    function setAttesterCommittee() external onlyOwner {
+        // Creates a new committee based on active attesters.
+        delete attesterCommittee;
```

Comment:
This is a lot of storage writes... 

Why do you need `attesterCommittee` and `validatorCommittee` in storage at all? It is not used anywhere. While the server can just picked up addresses from `nodes` function and that's it

Reply:
Follow up here: https://github.com/matter-labs/era-contracts/pull/555/files#r1705431453